src/export/InstanceExporter.ts

```diff
@@ -279,7 +279,11 @@ export class InstanceExporter implements Fishable {
           // against the correct choice slice
           if (
             instanceChild != null &&
-            (instance._sliceName ? choiceSlice.sliceName === instance._sliceName : true)
+            (instance._sliceName
+              ? parseFSHPath(choiceSlice.id).some(pathPart => {
+                  return pathPart.base.split(':')[1] === instance._sliceName;
```

Comment:
I think you can simplify this a little if you use `splitOnPathPeriods` instead of `parseFSHPath` here. Also, rather than checking the whole path with `some`, I think you always want to see the matching slice name on the second-to-last part of the path? The last part of the path is the choice element, so the part right before that should correspond to the element that `instance` currently represents. I'm not 100% sure about that, though. I tried to verify this by putting together a profile that had a choice element with multiple slice elements along its ancestry (for example, a path like `something[Blue].somethingElse[Red].value[x]`), but I haven't found a good starting resource type for such a profile yet.